recipes/gcc/all/conanfile.py

```diff
@@ -1,126 +1,162 @@
+from conan.tools.env import VirtualBuildEnv
 import os
 
-required_conan_version = ">=1.29.1"
+required_conan_version = ">=1.53.0"
 
 
 class GccConan(ConanFile):
     name = "gcc"
-    description = "The GNU Compiler Collection includes front ends for C, " \
-                  "C++, Objective-C, Fortran, Ada, Go, and D, as well as " \
-                  "libraries for these languages (libstdc++,...). "
+    description = (
+        "The GNU Compiler Collection includes front ends for C, "
+        "C++, Objective-C, Fortran, Ada, Go, and D, as well as "
+        "libraries for these languages (libstdc++,...). "
+    )
     topics = ("gcc", "gnu", "compiler", "c", "c++")
     homepage = "https://gcc.gnu.org"
     url = "https://github.com/conan-io/conan-center-index"
     license = "GPL-3.0-only"
     settings = "os", "compiler", "arch", "build_type"
-    _autotools = None
 
-    def build_requirements(self):
-        self.build_requires("flex/2.6.4")
-
-    def _configure_autotools(self):
-        if self._autotools:
-            return self._autotools
-        self._autotools = AutoToolsBuildEnvironment(self)
-        pkgversion = 'conan GCC %s' % self.version
-        bugurl = self.url + '/issues'
-        libdir = "%s/lib/gcc/%s" % (self.package_folder, self.version)
-        args = [
-            "--enable-languages=c,c++",
-            "--disable-nls",
-            "--disable-multilib",
-            "--disable-bootstrap",
-            "--with-system-zlib",
-            "--with-gmp=%s" % self.deps_cpp_info['gmp'].rootpath,
-            '--with-mpc=%s' % self.deps_cpp_info["mpc"].rootpath,
-            "--with-mpfr=%s" % self.deps_cpp_info["mpfr"].rootpath,
-            "--without-isl",
-            "--libdir=%s" % libdir,
-            '--with-pkgversion=%s' % pkgversion,
-            "--program-suffix=-%s" % self.version,
-            "--with-bugurl=%s" % bugurl
-        ]
-        if self.settings.os == "Macos":
-            xcrun = tools.XCRun(self.settings)
-            args.extend([
-                '--with-native-system-header-dir=/usr/include',
-                "--with-sysroot={}".format(xcrun.sdk_path)
-            ])
-        self._autotools.libs = []  # otherwise causes config.log to fail finding -lmpc
+    def configure(self):
         if self.settings.compiler in ["clang", "apple-clang"]:
-            # xgcc: error: unrecognized command-line option -stdlib=libc++
-            if self.settings.compiler.libcxx == "libc++":
-                self._autotools.cxx_flags.remove("-stdlib=libc++")
-            elif self.settings.compiler.libcxx in ["libstdc++", "libstdc++11"]:
-                self._autotools.cxx_flags.remove("-stdlib=libstdc++")
-        self._autotools.configure(args=args, configure_dir=self._source_subfolder)
-        return self._autotools
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+            # Can't remove this from cxxflags with autotools - so get rid of it
```

Comment:
should there be a validate for consumer to use the same one?

Reply:
I've removed this because if `compiler.libcxx != None`, `-stdlib=libstdc++` gets propagated to `cxxflags` in autotools, which breaks a clang compilation. Removing this means that `-stdlib=libstdc++` doesn't get added to `cxxflags` and the build will succeed.

Wrt validation, I guess the question is - which of the following options do you prefer:
1. force the user to specify `compiler.libcxx=None` in their profile (or CLI override) and trigger an exception in `validate()` if they haven't done this and `compiler.libcxx` has a value
2. silently fix the issue by deleting `compiler.libcxx` when clang is used and progress the build (the current solution)

I think my preference here is (2), since it provides the least friction to the user. Note that `settings.compiler` doesn't form part of the `package_id` calculation, so the package id will be the same regardless of the `libcxx` value.

IMO, (1) would be a little strange to me as a user because it requires me to know something very specific about this recipe, especially when there's only really one right answer, and not a choice that's presented to the user.

Thoughts?

Reply:
That's a good argument, you can the libcxx that version of gcc gives.
I accept